dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -192,6 +193,7 @@ void PushingToViewsBlockOutputStream::process(const Block & block, size_t view_n
         /// and two-level aggregation is triggered).
         in = std::make_shared<SquashingBlockInputStream>(
             in, context.getSettingsRef().min_insert_block_size_rows, context.getSettingsRef().min_insert_block_size_bytes);
+        in = std::make_shared<ConvertingBlockInputStream>(context, in, view.out->getHeader(), ConvertingBlockInputStream::MatchColumnsMode::Position);
```

Comment:
Can't figure out why `Position` mode is used instead of `Name`. Simple example:
`create table t1 (a UInt8, b String) engine MergeTree ORDER by a;
create table t2 (c UInt8, d String) engine MergeTree ORDER BY c;
create materialized view t3 to t1 (b String, a UInt8) AS SELECT d as b, c as a FROM t2;
insert into t2 values (1, 'test');`
`Received exception from server (version 19.13.1):
Code: 6. DB::Exception: Received from localhost:9000. DB::Exception: Cannot parse string 'test' as UInt8: syntax error at begin of string. Note: there are toUInt8OrZero and toUInt8OrNull functions, which returns zero/NULL instead of throwing exception.: while converting source column b to destination column a: while pushing to view default.t3.`